Mathlib/Analysis/NormedSpace/Connected.lean

```diff
@@ -122,6 +124,47 @@ section NormedSpace
 
 variable {E : Type*} [NormedAddCommGroup E] [NormedSpace ℝ E]
 
+section Ball
+
+namespace Metric
+
+theorem ball_contracitble {x : E} {r : ℝ} (hr : 0 < r) :
+    ContractibleSpace (ball x r) :=
+  Convex.contractibleSpace (convex_ball _ _) (by simpa)
+
+theorem eball_contractible {x : E} {r : ENNReal} (hr : 0 < r) :
+    ContractibleSpace (EMetric.ball x r) := by
+  cases r with
+  | top =>
+    rw [eball_top_eq_univ, (Homeomorph.Set.univ E).contractibleSpace_iff]
+    exact RealTopologicalVectorSpace.contractibleSpace
+  | coe r =>
+    rw [emetric_ball_nnreal]
+    apply ball_contracitble
+    simpa using hr
+
+theorem ball_PathConnected {x : E} {r : ℝ} (hr : 0 < r) :
```

Comment:
This one should definitely be named differently: probably "ball_pathConnectedSpace".